src/Cake.Core/IO/ProcessRunner.cs

```diff
@@ -123,9 +126,12 @@ internal ProcessStartInfo GetProcessStartInfo(FilePath filePath, ProcessSettings
 
             if (!settings.Silent)
             {
-                // Log the filename and arguments.
-                var message = string.Concat(fileName, " ", arguments.RenderSafe().TrimEnd());
-                _log.Verbose(Verbosity.Diagnostic, "Executing: {0}", message);
+                using (_log.WithVerbosity(_showCommandLine ? Verbosity.Diagnostic : _log.Verbosity))
+                {
+                    // Log the filename and arguments.
+                    var message = string.Concat(fileName, " ", arguments.RenderSafe().TrimEnd());
+                    _log.Verbose(Verbosity.Diagnostic, "Executing: {0}", message);
```

Comment:
Is using WithVerbosity needed?

Wouldn't if be enough to 

```csharp
_log.Verbose(_showCommandLine  ? _log.Verbosity : Verbosity.Diagnostic,
    "Executing: {0}", message);
```

Reply:
Oh yeah. Doh!